mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -60,26 +59,37 @@ def _hit_to_mlflow_experiment(self, hit: Any) -> Experiment:
                           artifact_location=hit.artifact_location,
                           lifecycle_stage=hit.lifecycle_stage)
 
-    def _hit_to_mlflow_run(self, hit: Any) -> Run:
+    def _hit_to_mlflow_run(self, hit: Any, columns_to_whitelist_key_dict: dict = None) -> Run:
         return Run(run_info=self._hit_to_mlflow_run_info(hit),
-                   run_data=self._hit_to_mlflow_run_data(hit))
+                   run_data=self._hit_to_mlflow_run_data(hit._source,
+                                                         columns_to_whitelist_key_dict))
 
     def _hit_to_mlflow_run_info(self, hit: Any) -> RunInfo:
-        return RunInfo(run_uuid=hit.meta.id, run_id=hit.meta.id,
-                       experiment_id=str(hit.experiment_id),
-                       user_id=hit.user_id,
-                       status=hit.status,
-                       start_time=hit.start_time,
-                       end_time=hit.end_time if hasattr(hit, 'end_time') else None,
-                       lifecycle_stage=hit.lifecycle_stage, artifact_uri=hit.artifact_uri)
-
-    def _hit_to_mlflow_run_data(self, hit: Any) -> RunData:
-        return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in
-                                (hit.latest_metrics if hasattr(hit, 'latest_metrics') else [])],
-                       params=[self._hit_to_mlflow_param(p) for p in
-                               (hit.params if hasattr(hit, 'params') else [])],
-                       tags=[self._hit_to_mlflow_tag(t) for t in
-                             (hit.tags if hasattr(hit, 'tags') else[])])
+        return RunInfo(run_uuid=hit._id, run_id=hit._id,
+                       experiment_id=str(hit._source.experiment_id),
+                       user_id=hit._source.user_id,
+                       status=hit._source.status,
+                       start_time=hit._source.start_time,
+                       end_time=hit._source.end_time if hasattr(hit._source, 'end_time') else None,
+                       lifecycle_stage=hit._source.lifecycle_stage if
+                       hasattr(hit["_source"], 'lifecycle_stage') else None,
+                       artifact_uri=hit._source.artifact_uri
+                       if hasattr(hit["_source"], 'artifact_uri') else None)
```

Comment:
Why do you sometimes access _source as if it was a key in a dict, and some other times as if it was a member of an object ? 
Also you used to access fields directly in the hit object, does the _source thing come from the elasticsearch version change ?

Reply:
Euh I don't know, it is fixed